virtual_rainforest/soil/carbon.py

```diff
@@ -0,0 +1,203 @@
+
+        Args:
+            pH: pH values for each soil grid cell
+            bulk_density: bulk density values for each soil grid cell
+            soil_moisture: soil moisture for each soil grid cell
+            soil_temp: soil temperature for each soil grid cell
+            percent_clay: Percentage clay for each soil grid cell
+        """
+
+        # This expression is drawn from (Mayes et al. (2012))
+        binding_affinity = 10.0 ** (
+            BINDING_WITH_PH["slope"] * pH + BINDING_WITH_PH["intercept"]
+        )
+
+        # This expression is also drawn from Mayes et al. (2012)
+        # Original paper also depends on Fe concentration, but we are ignoring this for
+        # now
+        Q_max = bulk_density * 10 ** (
+            MAX_SORPTION_WITH_CLAY["slope"] * np.log10(percent_clay)
+            + MAX_SORPTION_WITH_CLAY["intercept"]
+        )
+
+        # Using the above calculate the equilibrium MAOM pool
+        equib_maom = (binding_affinity * Q_max * self.lmwc) / (
+            1 + self.lmwc * binding_affinity
+        )
+
+        # Find scalar factors that multiple rates
+        temp_scalar = scalar_temperature(soil_temp)
+        moist_scalar = scalar_moisture(soil_moisture)
+
+        flux = temp_scalar * moist_scalar * self.lmwc * (equib_maom - self.maom) / Q_max
+
+        return -flux, flux
+
+
+def scalar_temperature(soil_temp: NDArray[np.float32]) -> NDArray[np.float32]:
+    """Convert soil temperature into a factor to multiply rates by.
+
+    This form is used in Abramoff et al. (2018) to minimise differences with the
+    CENTURY model. We very likely want to define our own functional form here. I'm
+    also a bit unsure how this form was even obtained, so further work here is very
+    needed.
+
+    Args:
+       soil_temp: soil temperature for each soil grid cell
+    """
+
+    # This expression is drawn from Abramoff et al. (2018)
+    numerator = TEMP_SCALAR["t_2"] + (TEMP_SCALAR["t_3"] / pi) * np.arctan(
+        pi * (soil_temp - TEMP_SCALAR["t_1"])
+    )
+    denominator = TEMP_SCALAR["t_2"] + (TEMP_SCALAR["t_3"] / pi) * np.arctan(
+        pi * TEMP_SCALAR["t_4"] * (TEMP_SCALAR["ref_temp"] - TEMP_SCALAR["t_1"])
+    )
+
+    return np.divide(numerator, denominator)
+
+
+def scalar_moisture(soil_moisture: NDArray[np.float32]) -> NDArray[np.float32]:
+    """Convert soil moisture into a factor to multiply rates by.
+
+    This form is used in Abramoff et al. (2018) to minimise differences with the
+    CENTURY model. We very likely want to define our own functional form here. I'm
+    also a bit unsure how this form was even obtained, so further work here is very
+    needed.
+
+    Args:
+        soil_moisture: soil moisture for each soil grid cell
+    """
+
+    # This expression is drawn from Abramoff et al. (2018)
+    scaler = 1 / (
+        1
+        + MOISTURE_SCALAR["coefficient"]
+        * e ** (-MOISTURE_SCALAR["exponent"] * soil_moisture)
+    )
+
+    return scaler
```

Comment:
Two things:

1. I'd just return the expression directly without the temporary variable
2. Use `np.exp()` to calculate the exponent rather than `e ** x` (I know these things look the same but I *think* a separate `exp()` function is usually provided by programming languages because it's either faster or more accurate to do it that way)

```suggestion
    return 1 / (
        1
        + MOISTURE_SCALAR["coefficient"]
        * np.exp(-MOISTURE_SCALAR["exponent"] * soil_moisture)
    )
```